tree-walk/src/interpreter.cpp

```diff
@@ -279,6 +379,27 @@ Value InterpreterVisitor::operator()(const Literal &ltrl) {
   }
 }
 
+Value InterpreterVisitor::operator()(const Set &set) {
+  Value obj = std::visit(*this, *set.object);
+  if (!std::holds_alternative<ClsInstShrdPtr>(obj)) {
+    throw InterpretException("Tried to set a property on non class instance " +
+                             std::visit(s_valuePrinter, obj));
+  }
+
+  Value val = std::visit(*this, *set.value);
+  // Create a copy of the function and rename it
```

Comment:
Move comment into block. Also consider the renaming to FunctionMetadata or similar. FunctionInstance? and FunctionCore?